src/metrics/matcher/namespace/resolver.go

```diff
@@ -0,0 +1,46 @@
+// Package namespace provides namespace functionality for the matcher.
```

Comment:
needed to separate out a shared namespace resolver for the cache and non-cache matchers to use, since they are in separate pkgs.